ultralytics/data/explorer/explorer.py

```diff
@@ -466,3 +469,136 @@ def generate_report(self, result):
         TODO
         """
         pass
+
+    def count_labels(self, labels: list):
+        """
+        Count occurrences of given labels in dataset.
+
+        Args:
+            labels (list): List of labels to count their occurrences.
+
+        Returns:
+            filtered_df (pandas.DataFrame): Dataframe containing records containing given labels.
+            label_counts (dict): Dictionary containing labels with their count.
+
+        Example:
+            ```python
+            exp = Explorer()
+            exp.create_embeddings_table()
+            filtered_data, label_counts = exp.count_labels(labels)
+            ```
+        """
+        label_counts = {}
+        df = self.table.to_pandas()
+        filtered_df = df[df["labels"].apply(lambda l: bool(set(l) & set(labels)))]
+        flatten_labels = [l for ls in list(filtered_df.labels) for l in ls]
+        for label in labels:
+            if label in flatten_labels:
+                label_counts[label] = flatten_labels.count(label)
+            else:
+                label_counts[label] = 0
+        return filtered_df, label_counts
+
+    def plot_count_labels(self, labels: list) -> Image.Image:
+        """
+        Plot occurrences of given labels in dataset.
+
+        Args:
+            labels (list): List of labels to count their occurrences.
+
+        Returns:
+            (PIL.Image): Image containing the plot.
+
+        Example:
+            ```python
+            exp = Explorer()
+            exp.create_embeddings_table()
+            img = exp.plot_count_labels(labels)
+            ```
+        """
+        filtered_data, labels_count = self.count_labels(labels)
+        print(filtered_data)
+        if len(filtered_data) == 0:
+            LOGGER.info("No results found.")
+            return None
+        filtered_data = pa.Table.from_pandas(filtered_data)
+        img = plot_query_result(filtered_data)
+        return Image.fromarray(img)
+
+    def count_dataset_labels(self) -> dict:
+        """
+        Count of Occurrences of all labels in dataset.
+
+        Returns:
+            label_counts (dict): Dictionary containing labels with their count.
+
+        Example:
+            ```python
+            exp = Explorer()
+            exp.create_embeddings_table()
+            label_count = exp.dataset_label_count()
+            ```
+        """
+        label_counts = {}
+        dataset_labels = self.table.to_lance().to_table(columns=["labels"]).to_pandas()
+        flatten_labels = [l for ls in list(dataset_labels) for l in ls]
+        unique_labels = list(set(flatten_labels))
+
+        for label in unique_labels:
+            label_counts[label] = flatten_labels.count(label)
+        return label_counts
+
+    def unique_labels(self):
```

Comment:
Isn't unique labels just the list of all classes of the dataset? this info is already present in the dataset yaml I think.. Or is this something else

Reply:
Here Unique Labels shows, that all labels are available in images, whereas in yaml all the labels that can be in the dataset are present.

This function helps to compare the datasets labels and labels present in images